executor/analyze.go

```diff
@@ -1160,30 +1160,8 @@ func (e *AnalyzeFastExec) buildStats() (hists []*statistics.Histogram, cms []*st
 	}
 
 	defer e.job.Update(int64(e.rowCount))
-
-	// If total row count of the table is smaller than 2*MaxSampleSize, we
-	// translate all the sample tasks to scan tasks.
-	sampleSize := e.opts[ast.AnalyzeOptNumSamples]
-	if e.rowCount < sampleSize*2 {
-		for _, task := range e.sampTasks {
-			e.scanTasks = append(e.scanTasks, task.Location)
-		}
-		e.sampTasks = e.sampTasks[:0]
-		e.rowCount = 0
-		return e.runTasks()
-	}
-
-	randPos := make([]uint64, 0, sampleSize+1)
-	for i := 0; i < int(sampleSize); i++ {
-		randPos = append(randPos, uint64(rander.Int63n(int64(e.rowCount))))
-	}
-	sort.Slice(randPos, func(i, j int) bool { return randPos[i] < randPos[j] })
-
-	for _, task := range e.sampTasks {
-		begin := sort.Search(len(randPos), func(i int) bool { return randPos[i] >= task.BeginOffset })
-		end := sort.Search(len(randPos), func(i int) bool { return randPos[i] >= task.EndOffset })
-		task.SampSize = uint64(end - begin)
-	}
+	totalSampSize := e.opts[ast.AnalyzeOptNumSamples]
+	e.estSampStep = uint32(e.rowCount / totalSampSize)
```

Comment:
The `e.rowCount` is sum `mvcc.num_rows`. It contains many different mvcc version for one row. Do you make sure the `estSampStep` could handle this situation?

Reply:
By the way, the `estSampStep` filters certain rows for a not changing table. Could we add some rand for it?

Reply:
> The `e.rowCount` is sum `mvcc.num_rows`. It contains many different mvcc version for one row. Do you make sure the `estSampStep` could handle this situation?

@lzmhhh123 @coocood Do you have any suggestions to deal with this problem?

Reply:
Maybe we can make `estSampStep` scaling by `region.mvcc.num_rows / region.num_rows` on tikv?

Reply:
The `mvcc.num_rows` only contains one version for each key.

Reply:
`mvcc.num_versions` constains all the versions in the region.